service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -570,7 +571,7 @@ public ReportWithoutLibrary createReportWithoutLibrary(
     newReport.setNumberOfBooks(req.getNumberOfBooks());
     newReport.setMostRecentShipmentYear(req.getMostRecentShipmentYear());
     newReport.setHasSpace(req.getHasSpace());
-    newReport.setCurrentStatus(req.getCurrentStatus());
+    newReport.setCurrentStatus(req.getCurrentStatus().toArray(new String[0]));
```

Comment:
It looks like `setCurrentStatus` takes in varargs, and IntelliJ is giving a warning that this could be interpreted differently from what you expect. Double check that this is getting interpreted correctly.

Reply:
Not super familiar but is there a spread operator equivalent in Java? I know you can use reflection and .invoke(args) but thats a little too ugly for this no? Can you think of a better solution? I'm no good at Java..

Reply:
I think it might be fine if you just pass in the original `Object[]` from `req.getCurrentStatus()`. I think casting to a `String[]` instead of an `Object[]` here makes it think that the array might be a single param.

Reply:
you are a java god (and yes thats a compliment)